### PR TITLE
Restrict EPG delete with active EPs

### PR DESCRIPTION
**Pull request by DivyaVavili**
PR includes:
- State change in EPG to maintain active EP count and restrict deletion when active EPs are present
- Fix bug in network deletion on active EPs (the operation was partially cleaning up tenant state)
- Fix some minor typo

### CODE REVIEW

**Comment by gaurav-dalvi** [comment]
LGTM

File: netmaster/master/endpointGroup.go [hunk 1]
```diff
@@ -149,6 +150,10 @@ func DeleteEndpointGroup(tenantName, groupName string) error {
 		return err
 	}
 
+	if epgCfg.EpCount != 0 {
+		return core.Errorf("Error: EPG has active endpoints")
```
Comment (abhinandanpb) [inline netmaster/master/endpointGroup.go]: can we mention the epg name here ?  So while debugging we can use epg inspect to list the active endpoints.

File: netmaster/objApi/apiController.go [hunk 2]
```diff
@@ -667,8 +670,12 @@ func (ac *APIController) EndpointGroupDelete(endpointGroup *contivModel.Endpoint
 			endpointGroup.GroupName, endpointGroup.Links.AppProfile.ObjKey)
 	}
 
-	endpointGroupCleanup(endpointGroup)
-	return nil
+	err := endpointGroupCleanup(endpointGroup)
+	if err != nil {
+		log.Errorf("EPG cleanup failed: %+v", err)
```
Comment (abhinandanpb) [inline netmaster/objApi/apiController.go]: same as above.

File: systemtests/network_test.go [hunk 3]
```diff
@@ -266,6 +266,51 @@ func (s *systemtestSuite) testNetworkAddDeleteNoGateway(c *C, encap string) {
...
+func (s *systemtestSuite) TestNetworkEPGAddDelete(c *C) {
+
+	var err error
+	numContainers := 4
+	networkName := "net1"
+	epgName := "epg1"
+
+	for i := 0; i < s.iterations; i++ {
+		// Create Endpoints
+		network := &client.Network{
```
Comment (abhinandanpb) [inline systemtests/network_test.go]: Can we make this multiple networks , multiple epgs ?

Comment (abhinandanpb) [inline systemtests/network_test.go]: Also should this test be run in L3 mode as well ? If so we need to add setup bgp step

File: netmaster/master/endpointGroup.go (hunk 1 above)
Comment (DivyaVavili) [inline netmaster/master/endpointGroup.go]: Adding a EPG name here

File: netmaster/objApi/apiController.go (hunk 2 above)
Comment (DivyaVavili) [inline netmaster/objApi/apiController.go]: The previous return will be printed here. So, it should contain the EPG name as well

File: systemtests/network_test.go (hunk 3 above)
Comment (DivyaVavili) [inline systemtests/network_test.go]: I'll add support for multiple EPGs, but not sure if adding the multiple networks will help here and might just end up increasing test execution time. That part is already been tested in "TestNetworkAddDelete*" routines. Does adding/deleting EPG have a different effect in L3 mode? If so, I'll add support for that.

**Pull request merged**